speedtest/datadog_checks/speedtest/speedtest.py

```diff
@@ -0,0 +1,163 @@
+import json
+import subprocess
+from typing import Any, Dict
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+MEGABYTE_TO_MEBIBYTE = 1048576.0 / 1000000.0
+
+
+class SpeedtestCheck(AgentCheck):
+    def check(self, *args, **kwargs):
+        # type: (Dict[str, Any]) -> None
+        host = self.instance.get("host")
+        ip = self.instance.get("ip")
+        interface = self.instance.get("interface")
+        server_id = self.instance.get("server_id")
+        tags = self.instance.get("tags") or []
+
+        # test config for only **one** of the following above options
+        c = 0
+        for opt in (host, ip, interface, server_id):
+            c += int(opt not in [None, ""])  # trick, int(bool) returns 0 if False, 1 if True
+        if c > 1:
+            raise ConfigurationError("Only one of `host`, `ip`, `interface` or `server_id` may be configured.")
+
+        # Build command
+        cmd = self._build_command(host, ip, interface, server_id)
+
+        # call to speedtest-cli on the system path
+        try:
+            payload = self._call_command(cmd)
+            self._submit_data(payload, tags)
+            self.service_check("speedtest.run", self.OK)
+        except Exception as e:
+            # something went wrong with this command
+            self.log.exception(e)
+            self.service_check("speedtest.run", self.CRITICAL, message=str(e))
+
+    def _build_command(self, host, ip, interface, server_id):
+        # Build command
+        cmd = "speedtest-cli -f json -p no -A -P 8 "
+        if host:
+            cmd += " --host={}".format(host)
+        if ip:
+            cmd += " --ip={}".format(ip)
+        if interface:
+            cmd += " --interface={}".format(interface)
+        if server_id:
+            cmd += " --server-id={}".format(server_id)
+
+        return cmd
+
+    def _call_command(self, cmd):
+        # we keep this private so we can mock this in tests
+        result = subprocess.check_output(cmd, stdout=subprocess.PIPE, shell=True)
```

Comment:
When running this check on an Ubuntu machine with Agent 7.19.2, the following error gets logged:

```
2020-06-17 14:40:29 UTC | CORE | ERROR | (pkg/collector/python/datadog_agent.go:116 in LogMessage) | speedtest:8987a6148e07ec77 | (speedtest.py:36) | stdout argument not allowed, it will be overridden.
Traceback (most recent call last):
  File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/speedtest/speedtest.py", line 31, in check
    payload = self._call_command(cmd)
  File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/speedtest/speedtest.py", line 55, in _call_command
    result = subprocess.check_output(cmd, stdout=subprocess.PIPE, shell=True)
  File "/opt/datadog-agent/embedded/lib/python3.8/subprocess.py", line 404, in check_output
    raise ValueError('stdout argument not allowed, it will be overridden.')
ValueError: stdout argument not allowed, it will be overridden.
```

Reply:
fixed in the latest push, was using `check_output(` which already handles this